op-interop-mon/flags/flags.go

```diff
@@ -0,0 +1,66 @@
+package flags
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/urfave/cli/v2"
+
+	opservice "github.com/ethereum-optimism/optimism/op-service"
+	oplog "github.com/ethereum-optimism/optimism/op-service/log"
+	opmetrics "github.com/ethereum-optimism/optimism/op-service/metrics"
+	"github.com/ethereum-optimism/optimism/op-service/oppprof"
+	oprpc "github.com/ethereum-optimism/optimism/op-service/rpc"
+)
+
+const EnvVarPrefix = "OP_INTEROP_MON"
+
+func prefixEnvVars(name string) []string {
+	return opservice.PrefixEnvVar(EnvVarPrefix, name)
+}
+
+var (
+	// Required Flags
+	L2RpcsFlag = &cli.StringSliceFlag{
+		Name:     "l2-rpcs",
+		Usage:    "The RPC URLs for the L2 chains to monitor",
+		EnvVars:  prefixEnvVars("L2_RPCS"),
+		Required: true,
+	}
+
+	// Optional Flags
+	PollIntervalFlag = &cli.DurationFlag{
+		Name:    "poll-interval",
+		Usage:   "How frequently to poll L2 for new blocks",
+		Value:   12 * time.Second,
+		EnvVars: prefixEnvVars("POLL_INTERVAL"),
+	}
+)
+
+var requiredFlags = []cli.Flag{
+	L2RpcsFlag,
+}
+
+var optionalFlags = []cli.Flag{
+	PollIntervalFlag,
+}
+
+func init() {
+	optionalFlags = append(optionalFlags, oprpc.CLIFlags(EnvVarPrefix)...)
+	optionalFlags = append(optionalFlags, oplog.CLIFlags(EnvVarPrefix)...)
+	optionalFlags = append(optionalFlags, opmetrics.CLIFlags(EnvVarPrefix)...)
+	optionalFlags = append(optionalFlags, oppprof.CLIFlags(EnvVarPrefix)...)
+
+	Flags = append(requiredFlags, optionalFlags...)
+}
+
+var Flags []cli.Flag
+
+func CheckRequired(ctx *cli.Context) error {
+	for _, f := range requiredFlags {
+		if !ctx.IsSet(f.Names()[0]) {
+			return fmt.Errorf("flag %s is required", f.Names()[0])
+		}
+	}
+	return nil
+}
```

Comment:
I'm surprised we need this, it could probably be removed.

Reply:
Actually, this is in use. Main checks that all required flags are set before running. RPCs is currently required.